src/types/components.ts

```diff
@@ -18,4 +18,9 @@ export type Country = {
 };
 
 //currency selector
-export type Currency = { code: string; name?: string; min?: number };
+export type Currency = {
+  code: string;
+  name?: string;
+  min?: number;
+  rate: number | null;
```

Comment:
```suggestion
  rate?: number;
```

Reply:
had I added `rate` as optional property, I would have to manually search throughout where `type Currency` is used and added `rate` accordingly which is easy to miss. 

adding `rate` as explicit `number | null` - suddenly typechecker is so helpful
<img width="866" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/ceef8d51-7ee5-4c6e-8940-60d6215db166">

there's no way to miss where `rate` is needed - transform values, default values etc.

Reply:
That was a useful and once you were done adding the missing property, you could've safely made `rate` optional again.
Leaving it as a union type only forces verbose code.

Reply:
@misicnenad, I used the video because I thought you need more context on the difference between `rate?:` and `rate: x | undefined` regarding the use of it in this PR. 

> Leaving it as a union type only forces verbose code.
Again, on the context of what's being done here - I would argue it's not even verbose - only the wise api needs an explicit `null`. most of the consumers need to add it anyway. 

On the otherhand, I would have considered `rate?:` if most of the consumers doesn't need to set it.

Reply:
@misicnenad 
> I think you too understand that this could now safely be made an optional type and you also know that it wouldn't take more than a minute of your time to make the change

So we just temp opt for a type variation and once satisfied we add another effort to revert to a more lax variant? Is this how you propose to maintain our interfaces? 

> just as I knew in similar situations when you asked me to make trivial changes in my own PRs,

The difference between you and me is that I don't persist on trivial things. When I mark my comment as `nit` I don't go around forcing other developer to make that change to suit my style as it is more of a suggestion. I don't understand why you are going so much with this trivial `number | null` choice even I explained to you the reason I did in the first place.

I assume it came from when you unneccesarily changed `DonaterConfig | null` to `DonaterConfig?` which if you haven't done in the first place, would have not caught my attention. Futhermore, I didn't relentlessly force you to change it - you just reverted it on your own with some suble insult
> I guess this is just a matter of preference, so will revert the change so as not to hold up the whole PR for a trifle.

If you are making changes for the sake of personal preference - probably its better not to change it in the first place?


> that's why I no longer rename your classes: string props into className, even though I prefer that name; 

Again, what benefit would you get from doing so? - so you're doing yourself a good favor on leaving it be.

> that's why I'd ask you to reconsider your position on this topic and make this an optional type, at least as a sign of goodwill.

If what you are suggesting is objective and has effect on the workings of the web-app, I would have accepted this in no time - but suggesting something subjective (because you can't stand seeing a single `null` ? ) How do you expect me to follow this? 

> Otherwise we're just destined to continue the cycle. The result? PRs will be halted for silly reasons, resentment will build, time (and nerves) will be wasted, code quality will drop. It doesn't have to be this way.

Take a look back, when you saw `number | null` you could have just let it go as there's really no substance into it. But you did push on it - I provided with you an objective reason for such decision and you go about finding more holes into it further pushing this futile argument to consume more of our time.

Reply:
@ap-justin 

> I assume it came from when you unnecessarily changed DonaterConfig | null to DonaterConfig? which if you haven't done in the first place, would have not caught my attention. Futhermore, I didn't relentlessly force you to change it - you just reverted it on your own with some suble insult
> 
> > I guess this is just a matter of preference, so will revert the change so as not to hold up the whole PR for a trifle.

This wasn't meant as any type of insult and I'm sorry that you took it that way. I reverted the change, as you said - "on my own", because I didn't want to create *this* type of discussion for nothing/for a trifle.

> I provided with you an objective reason for such decision and you go about finding more holes into it further pushing this futile argument to consume more of our time.

You may be right about me pushing. Occasionally I find your reasons misguided or unconvincing, and especially so after I feel (rightly or wrongly) like you're ignoring the "holes" that I find in them - this causes me to get frustrated, so I push you to address the "holes" I had found. 

I'm sorry our relations are so low that we can't even have an honest discussion about how low they are; and I'm tired of pointing fingers and remembering every slight. I'm not excusing myself though, I contributed my fair share to the current state of things. Maybe the cause of us consistently misunderstanding each other is the baggage our relationship carries.

You're a good guy Justin, I can hear and see it on our calls. You're also a good and dedicated dev, that I can see in our code. That's enough for me to continue working with you, despite our arguments.
I hope, with time, we get to a point where we at least interpret each other's intentions as well-meaning; that's the least colleagues can expect of each other.

I won't be holding the PR any longer over this, the changes are overall good.